### PR TITLE
fix(Script/nexus/EOE and Oculus) The Scion of Eternity(EOE) target wrong and player can not ride the dragon in Oculus

### PR DESCRIPTION
**Pull request by cctvzj**
## Issues Addressed:

- fix a issue #14837 
- another issue still no finding by other

## Tests Performed:

- Tested ingame

## How to Test the Changes:

1.go to EOE , attack Malygos enter p2 stage，when some players in a hover disk that The Scion of Eternity do not cast spell to them 
2. .go to Oculus,kill the Drakos  and get dragon summon item. summon a dragon and ride it.
3.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/Nexus/EyeOfEternity/boss_malygos.cpp [hunk 1]
```diff
@@ -1201,14 +1201,13 @@ class npc_scion_of_eternity : public CreatureScript
                             for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
                                 if (Player* pPlayer = i->GetSource())
                                 {
-                                    if (!pPlayer->IsAlive() || pPlayer->IsGameMaster())
-                                        continue;
+                                    if (pPlayer->IsAlive()&&!pPlayer->GetVehicle())
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Nexus/EyeOfEternity/boss_malygos.cpp:1204]: this if statement does nothing right now

File: src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp [hunk 2]
```diff
@@ -353,8 +354,8 @@ class npc_oculus_drake : public CreatureScript
         uint16 despawnTimer;
 
         void IsSummonedBy(WorldObject* summoner) override
-        {
-            if (summoner->GetTypeId() != TYPEID_UNIT)
+        {
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp:357]: remove blankspace

File: src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp [hunk 3]
```diff
@@ -392,9 +393,10 @@ class npc_oculus_drake : public CreatureScript
 
         void PassengerBoarded(Unit* passenger, int8 /*seatid*/, bool add) override
         {
+           
             if (passenger->GetTypeId() != TYPEID_PLAYER)
                 return;
-
+
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp:399]: same as above

File: src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp [hunk 4]
```diff
@@ -765,7 +767,7 @@ class spell_oculus_call_ruby_emerald_amber_drake : public SpellScriptLoader
         void SetDest(SpellDestination& dest)
         {
             // Adjust effect summon position
-            Position const offset = { 0.0f, 0.0f, 12.0f, 0.0f };
+            Position const offset = { 0.0f, 0.0f, 12.0f, 0.0f };  //12.0f
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp:770]: unnecessary comment

File: src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp [hunk 5]
```diff
@@ -30,6 +30,7 @@
 #include "Vehicle.h"
 #include <unordered_map>
 
+
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp:33]: remove the extra newline

File: src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp [hunk 6]
```diff
@@ -392,9 +393,9 @@ class npc_oculus_drake : public CreatureScript
 
         void PassengerBoarded(Unit* passenger, int8 /*seatid*/, bool add) override
         {
+           
             if (passenger->GetTypeId() != TYPEID_PLAYER)
                 return;
-
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp:397]: revert.

**Comment by Nyeriah** [comment]
Please address a single issue per PR, please don't stack unrelated issues in a single PR.

File: src/server/scripts/Northrend/Nexus/EyeOfEternity/boss_malygos.cpp [hunk 7]
```diff
@@ -1201,14 +1201,18 @@ class npc_scion_of_eternity : public CreatureScript
...
+                                        {
+                                            guids.push_back(pPlayer->GetGUID());
+                                        }
+                                    }
                                 }
                         if (!guids.empty())
                             if (Player* plr = ObjectAccessor::GetPlayer(*me, guids.at(urand(0, guids.size() - 1))))
                                 me->CastSpell(plr, SPELL_SCION_ARCANE_BARRAGE);
-
+                        guids.clear();
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Nexus/EyeOfEternity/boss_malygos.cpp:1215]: `guids` is a local variable, it only exists inside this scope, pointless to clear

**Pull request merged**